File: pset2/readability.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include "cs50.h"

int countLetter(string s);
int countWord(string s);
int countSentence(string s);

int main(void)
{
    string input = get_string("Text: ");

    int cntLetter = countLetter(input);
    int cntWord = countWord(input);
    int cntSentence = countSentence(input);
    float L = 100.0 * cntLetter / cntWord;
    float S = 100.0 * cntSentence / cntWord;
    int idx = (int)(0.0588 * L - 0.296 * S - 15.8 + 0.5);
    if (idx >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (idx < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", idx);
    }
}

int countLetter(string s)
{
    int cnt = 0;
    for (int i = 0; i < strlen(s); ++i)
    {
        if (islower(s[i]) || isupper(s[i]))
        {
            ++cnt;
        }
    }
    return cnt;
}

int countWord(string s)
{
    int cnt = 0;
    for (int i = 0; i < strlen(s); ++i)
    {
        if (s[i] == ' ')
        {
            ++cnt;
        }
    }
    return cnt + 1;
}


int countSentence(string s)
{
    int cnt = 0;
    for (int i = 0; i < strlen(s); ++i)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            ++cnt;
        }
    }
    return cnt;
}

File: pset6/readability/readability.py
import cs50


def main():
    input = cs50.get_string("Text: ")
    cntLetter = countLetter(input)
    cntWord = countWord(input)
    cntSentence = countSentence(input)
    L = 100.0 * cntLetter / cntWord
    S = 100.0 * cntSentence / cntWord
    idx = (int)(0.0588 * L - 0.296 * S - 15.8 + 0.5)
    if idx >= 16:
        print("Grade 16+")
    elif idx < 1:
        print("Before Grade 1")
    else:
        print("Grade %d" % idx)


def countLetter(s):
    cnt = 0
    for c in s:
        if c.isalpha():
            cnt += 1
    return cnt


def countWord(s):
    cnt = 0
    for c in s:
        if c == ' ':
            cnt += 1
    return cnt + 1


def countSentence(s):
    cnt = 0
    for c in s:
        if c == '.' or c == '!' or c == '?':
            cnt += 1
    return cnt


main()
